controllers/imagerepository_controller.go

```diff
@@ -161,6 +220,15 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}
 
+	// we are adding to map only for new provision, not for some partial actions,
+	// so report time only if time was recorded
+	provisionTime, timeRecorded := repositoryTimesForMetrics[componentIdForMetrics]
+	if timeRecorded {
+		imageRepositoryProvisionTimeMetric.Observe(time.Since(provisionTime).Seconds())
+	}
+	// remove component from metrics map
+	delete(repositoryTimesForMetrics, componentIdForMetrics)
```

Comment:
We can do the deletion only if the time was recorded.

Reply:
time will be always recorded at that point

Reply:
Then why do we need that `if` above?

Reply:
I mean in this case I guess I can just move that delete inside if